test/e2e/fleetautoscaler_test.go

```diff
@@ -0,0 +1,223 @@
+		return
+	}
+
+	// the fleet autoscaler should scale the fleet up now up to BufferSize
+	bufferSize := int32(fas.Spec.Policy.Buffer.BufferSize.IntValue())
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(bufferSize))
+	assert.Nil(t, err, "fleet did not sync with autoscaler")
+
+	// patch the autoscaler to increase MinReplicas and watch the fleet scale up
+	fas, err = patchFleetAutoscaler(fas, intstr.FromInt(int(bufferSize)), bufferSize+2, fas.Spec.Policy.Buffer.MaxReplicas)
+	assert.Nil(t, err, "could not patch fleetautoscaler")
+
+	bufferSize = int32(fas.Spec.Policy.Buffer.BufferSize.IntValue())
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(bufferSize))
+	assert.Nil(t, err, "fleet did not sync with autoscaler")
+
+	// patch the autoscaler to remove MinReplicas and watch the fleet scale down
+	fas, err = patchFleetAutoscaler(fas, intstr.FromInt(int(bufferSize)), 0, fas.Spec.Policy.Buffer.MaxReplicas)
+	assert.Nil(t, err, "could not patch fleetautoscaler")
+
+	bufferSize = int32(fas.Spec.Policy.Buffer.BufferSize.IntValue())
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(bufferSize))
+	assert.Nil(t, err, "fleet did not sync with autoscaler")
+
+	// do an allocation and watch the fleet scale up
+	fa := getAllocation(flt)
+	fa, err = alpha1.FleetAllocations(defaultNs).Create(fa)
+	assert.Nil(t, err)
+	assert.Equal(t, v1alpha1.Allocated, fa.Status.GameServer.Status.State)
+	err = framework.WaitForFleetCondition(flt, func(fleet *v1alpha1.Fleet) bool {
+		return fleet.Status.AllocatedReplicas == 1
+	})
+	assert.Nil(t, err)
+
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(bufferSize))
+	assert.Nil(t, err, "fleet did not sync with autoscaler")
+
+	// patch autoscaler to switch to relative buffer size and check if the fleet adjusts
+	_, err = patchFleetAutoscaler(fas, intstr.FromString("10%"), 1, fas.Spec.Policy.Buffer.MaxReplicas)
+	assert.Nil(t, err, "could not patch fleetautoscaler")
+
+	//10% with only one allocated GS means only one ready server
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(1))
+	assert.Nil(t, err, "fleet did not sync with autoscaler")
+
+	// delete the allocated GameServer and watch the fleet scale down
+	gp := int64(1)
+	err = alpha1.GameServers(defaultNs).Delete(fa.Status.GameServer.ObjectMeta.Name, &metav1.DeleteOptions{GracePeriodSeconds: &gp})
+	assert.Nil(t, err)
+	err = framework.WaitForFleetCondition(flt, func(fleet *v1alpha1.Fleet) bool {
+		return fleet.Status.AllocatedReplicas == 0 &&
+			fleet.Status.ReadyReplicas == 1 &&
+			fleet.Status.Replicas == 1
+	})
+	assert.Nil(t, err)
+}
+
+// TestAutoscalerStressCreate creates many fleetautoscalers with random values
+// to check if the creation validation works as expected and if the fleet scales
+// to the expected number of replicas (when the creation is valid)
+func TestAutoscalerStressCreate(t *testing.T) {
```

Comment:
:+1:

Reply:
Glad you like it :smiley: 
I am also planning to add a stress test for scaling during rolling updates, but in a separate PR.